### PR TITLE
Add filtering by status to travel pay claims

### PR DESCRIPTION
**Pull request by athifw**
Add filtering by status to travel pay claims

## Summary

- Displays all of the user's claims' statuses as checkboxes
- Claims are filtered when statuses are selected and "Apply filters" is clicked
- "Reset search" button selects all checkboxes and refreshes list of claims

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/83451

## Testing done

- Unit tests added
- Cypress tests added

## Screenshots

Screenshot 2024-06-17 at 1 39 40 PM

## What areas of the site does it impact?

Travel pay claims (`my-health/travel-claim-status`)

### CODE REVIEW

File: src/applications/travel-pay/tests/e2e/travel-pay.cypress.spec.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @department-of-veterans-affairs/axe-check-required */
```
[the remark this answers is not in the document]
Comment (athifw) [reply src/applications/travel-pay/tests/e2e/travel-pay.cypress.spec.js:1]: it looks like we do `cy.injectAxeThenAxeCheck();` in `beforeEach`

File: src/applications/travel-pay/containers/TravelPayStatusApp.jsx [hunk 2]
```diff
@@ -164,6 +203,47 @@ export default function App({ children }) {
                         />
                       </div>
                     </div>
+                    <va-accordion bordered open-single>
```
Comment (kjduensing) [inline src/applications/travel-pay/containers/TravelPayStatusApp.jsx:206]: Let's move the filter to its own component if possible.

File: src/applications/travel-pay/tests/containers/TravelPayStatusApp.unit.spec.jsx [hunk 3]
```diff
@@ -226,6 +226,29 @@ describe('App', () => {
     });
   });
 
+  it('displayed status filters correctly', async () => {
```
Comment (kjduensing) [inline src/applications/travel-pay/tests/containers/TravelPayStatusApp.unit.spec.jsx:229]: We need to make sure we include tests that cover [the acceptance criteria outlined in the ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/83451).

**Review by CBonade — DISMISSED** [review]

File: src/applications/travel-pay/components/TravelPayFilters.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { VaCheckbox } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+
+export default function TravelPayFilters(props) {
```
Comment (kjduensing) [inline src/applications/travel-pay/components/TravelPayFilters.jsx:5]: I'd love to see a separate component per filter. We can wrap the filter components in this filters component. Thoughts?

Comment (athifw) [reply src/applications/travel-pay/components/TravelPayFilters.jsx:5]: That's a great idea - I can make a new `TravelPayStatusFilters` component

File: src/applications/travel-pay/tests/e2e/travel-pay.cypress.spec.js [hunk 5]
```diff
@@ -55,4 +78,74 @@ describe(`${appName} -- Status Page`, () => {
...
+    cy.get('h2[data-testid="travel-claim-details"]')
+      .first()
+      .should('include.text', 'May 15, 2024');
+  });
+
+  it('preserves sort order when filters are applied', () => {
+    cy.get('select[name="claimsOrder"]').select('oldest');
+    cy.get('va-button[data-testid="Sort travel claims"]').click();
+    cy.openFiltersAndUncheckStatuses();
+    cy.selectVaCheckbox('SAVED_checkbox', true);
```
Comment (kjduensing) [inline src/applications/travel-pay/tests/e2e/travel-pay.cypress.spec.js:140]: I think I missed something - what is the `SAVED_checkbox`?

Comment (kjduensing) [reply src/applications/travel-pay/tests/e2e/travel-pay.cypress.spec.js:140]: I see that this test unchecks all the filters, which filters is it enabling in order to test that sort order is preserved?

**Review by kjduensing — DISMISSED** [review]
Looks good from my end 🚂

File: src/applications/travel-pay/components/TravelPayFilters.jsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      <va-accordion-item
+        bordered="true"
+        data-testid="filters-accordion"
+        header="Filter travel claims"
+      >
+        <div className="filter-your-results">
+          <TravelPayStatusFilters
+            statusesToFilterBy={statusesToFilterBy}
+            checkedStatuses={checkedStatuses}
+            statusFilterChange={statusFilterChange}
```
Comment (hinzed1127) [inline src/applications/travel-pay/components/TravelPayFilters.jsx:25]: small nit: [it's convention](https://developer.mozilla.org/en-US/docs/Web/Events/Event_handlers#using_onevent_properties) to use `onEvent` properties for event handlers. It just makes it more explicit that it's responding to an action.
```suggestion
            onStatusFilterChange={onStatusFilterChange}
```

File: src/applications/travel-pay/components/TravelPayStatusFilters.jsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      <p>Travel claim status</p>
+      {statusesToFilterBy.map((status, index) => (
+        <div key={index}>
+          <VaCheckbox
+            checked={checkedStatuses.includes(status)}
+            data-testid="status-filter"
+            name={`${status}_checkbox`}
+            key={status}
+            label={status}
+            onVaChange={e => statusFilterChange(e, status)}
```
Comment (hinzed1127) [inline src/applications/travel-pay/components/TravelPayStatusFilters.jsx:19]: same thing here to apply the aforementioned fix

**Review by hinzed1127 — DISMISSED** [review]
One small thing but looks good!

**Review by hinzed1127 — APPROVED** [review]
LGTM

**Comment by CBonade** [comment]
Platform review delayed due to failing e2e test stability review checks.

**Comment by athifw** [comment]
> Platform review delayed due to failing e2e test stability review checks.

Re-ran tests, seem to be passing now @CBonade

**Review by rsmithadhoc — APPROVED** [review]

**Pull request merged**